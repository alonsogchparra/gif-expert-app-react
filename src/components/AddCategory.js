import PropTypes from "prop-types";
import React, { useState } from "react";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <div className="form-group">
        <label for="inputValue">Insert the category that you looking for</label>
        <input
          id="inputValue"
          className="form-control form-control-lg"
          value={inputValue}
          type="text"
          onChange={handleInputChange}
        />
      </div>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
