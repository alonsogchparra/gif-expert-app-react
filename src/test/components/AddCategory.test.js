import { shallow } from "enzyme";
import React from "react";
import AddCategory from "../../components/AddCategory";

describe("Test on AddCategory", () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should show AddCategory correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should text box change", () => {
    const input = wrapper.find("input");
    const value = "Hello World";

    input.simulate("change", { target: { value } });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("input").prop("value")).toBe(value);
  });

  test("should not submit when text box is empty", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should submit post and call setCategories", () => {
    const input = wrapper.find("input");
    const value = "El Camino";

    input.simulate("change", { target: { value } });

    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toBeCalledWith(expect.any(Function));
  });
});
