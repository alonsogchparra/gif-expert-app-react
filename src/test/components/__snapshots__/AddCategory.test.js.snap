// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AddCategory should show AddCategory correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group mt-4"
  >
    <input
      className="form-control form-control-lg"
      id="inputValue"
      onChange={[Function]}
      placeholder="Insert the category"
      type="text"
      value=""
    />
  </div>
</form>
`;

exports[`Test on AddCategory should text box change 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group mt-4"
  >
    <input
      className="form-control form-control-lg"
      id="inputValue"
      onChange={[Function]}
      placeholder="Insert the category"
      type="text"
      value="Hello World"
    />
  </div>
</form>
`;
